eslint/cactbot-output-strings.js

```diff
@@ -19,10 +19,14 @@ const ruleModule = {
     messages: {
       noOutputStrings: 'no outputStrings in trigger',
       notFoundProperty: 'no \'{{prop}}\' in \'{{outputParam}}\'',
+      notFoundTemplate: '`output.{{prop}}(...)` doesn\'t have template \'{{template}}\'.',
+      missingTemplateValue: 'template \'{{prop}}\' is missing in function call',
+      tooManyParams: 'function `output.{{call}}()` take only {{num}} params',
```

Comment:
Does `notFoundTemplate` and `missingTemplateValue` already cover `tooManyParams`, and `tooManyParams` can be removed now? It seems like if you have 0 template parameters and you try to provide one, then you would get `notFoundTemplate`, and if you try to provide too many then you would get `missingTemplateValue`?

Reply:
I don't think so . If method is called `output.text({}, shouldNotExist)`

Reply:
Oh!  Thank you for the explanation.  I see what it's trying to do; it's trying to check the number of parameters to output.  I am unable to get the "zero" case to trigger.  Can you give me an example of incorrect text for zero?  I also think that it's ok to have 0 or 1 output properties for 0 template parameters (`output.text({})` is valid, I think?).

Can you try these triggers? You'll also need to fix code to handle missing ids, which currently throws an error in two places.
```
    {
      // This is valid.
      alarmText: (data, _, output) => output.text({}),

      // This is correctly an error.
      alertText: (data, _, output) => output.text(true, false),

      // These should be an error, but are not.
      infoText: (data, _, output) => output.text({ foo: 3 }),
      tts: (data, _, output) => output.text(true),

      outputStrings: {
        text: {
          en: 'text',
        },
      },
    },
    {
      // This is valid.
      alarmText: (data, _, output) => output.textWithParam({ param: 3 }),

      // This is correctly an error.
      alertText: (data, _, output) => output.textWithParam({}),

      // This should be an error but is not.
      infoText: (data, _, output) => output.textWithParam(),

      // This is an error, but has incorrect error text:
      // error  template 'param' is missing in function call               rulesdir/cactbot-output-strings
      // error  `output.param2(...)` doesn't have template '{{template}}'  rulesdir/cactbot-output-strings
      tts: (data, _, output) => output.textWithParam({ param2: 3 }),

      outputStrings: {
        textWithParam: {
          en: '${param}',
        },
      },
    },
```

I could also argue that typescript will handle the number and type of parameters eventually? Typescript will never know about template parameters so that's great to check, but it should eventually handle the number of parameters on its own when we convert trigger files and give `output` a type.

Reply:
yes, typescript should be able to check the number of args. But we haven't started migrating triggers to ts so I think it's necessary now.

Reply:
These two from my previous example still do not create errors.  Should they?
```
      infoText: (data, _, output) => output.text({ foo: 3 }),
      tts: (data, _, output) => output.text(true),
```

Also, I think grammar-wise, saying "take only 1 params" could be better.  I am still not sure why there is a zero case or how to trigger that, but maybe this could say: `'function \'output.{{call}}()\' takes only 1 parameter',` for the one case?